From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gregor MacLennan <gmaclennan@gmail.com>
Date: Tue, 28 Jul 2026 19:20:31 +0000
Subject: [PATCH 13/20] deps,cares: guard sys/random.h on iOS; android config
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

darwin ares_config.h guards HAVE_SYS_RANDOM_H to macOS (iOS SDK lacks it); android ares_config.h.

Co-authored-by: Andre Staltz <andre@staltz.com>
Co-authored-by: Julia Samól <j.samol@papers.ch>
Co-authored-by: Jaime Bernardo <jaime@janeasystems.com>
---
 deps/cares/config/android/ares_config.h | 17 ++++++++++-------
 deps/cares/config/darwin/ares_config.h  |  7 +++++++
 2 files changed, 17 insertions(+), 7 deletions(-)

diff --git a/deps/cares/config/android/ares_config.h b/deps/cares/config/android/ares_config.h
index d09f6af162..b3a649413b 100644
--- a/deps/cares/config/android/ares_config.h
+++ b/deps/cares/config/android/ares_config.h
@@ -20,7 +20,7 @@
 #define GETHOSTNAME_TYPE_ARG2 size_t
 
 /* Define to the type qualifier of arg 1 for getnameinfo. */
-#define GETNAMEINFO_QUAL_ARG1 
+#define GETNAMEINFO_QUAL_ARG1
 
 /* Define to the type of arg 1 for getnameinfo. */
 #define GETNAMEINFO_TYPE_ARG1 struct sockaddr *
@@ -35,10 +35,10 @@
 #define GETNAMEINFO_TYPE_ARG7 int
 
 /* Specifies the number of arguments to getservbyport_r */
-#define GETSERVBYPORT_R_ARGS 
+#define GETSERVBYPORT_R_ARGS
 
 /* Specifies the number of arguments to getservbyname_r */
-#define GETSERVBYNAME_R_ARGS 
+#define GETSERVBYNAME_R_ARGS
 
 /* Define to 1 if you have AF_INET6. */
 #define HAVE_AF_INET6
@@ -124,8 +124,11 @@
 /* Define to 1 if you have the getrandom function. */
 #define HAVE_GETRANDOM
 
+// nodejs-mobile patch: wrap with ifndef __ANDROID__
 /* Define to 1 if you have the getservbyport_r function. */
-/* #undef HAVE_GETSERVBYPORT_R */
+#ifndef __ANDROID__
+#define HAVE_GETSERVBYPORT_R 1
+#endif
 
 /* Define to 1 if you have the getservbyname_r function. */
 /* #undef HAVE_GETSERVBYNAME_R */
@@ -393,7 +396,7 @@
 #define CARES_RANDOM_FILE "/dev/urandom"
 
 /* Define to the type qualifier pointed by arg 5 for recvfrom. */
-#define RECVFROM_QUAL_ARG5 
+#define RECVFROM_QUAL_ARG5
 
 /* Define to the type of arg 1 for recvfrom. */
 #define RECVFROM_TYPE_ARG1 int
@@ -441,7 +444,7 @@
 #define RECV_TYPE_RETV ssize_t
 
 /* Define to the type qualifier of arg 2 for send. */
-#define SEND_QUAL_ARG2 
+#define SEND_QUAL_ARG2
 
 /* Define to the type of arg 1 for send. */
 #define SEND_TYPE_ARG1 int
@@ -476,4 +479,4 @@
 /* Define to 1 if pthread_init() exists */
 /* #undef HAVE_PTHREAD_INIT */
 
-    
+
diff --git a/deps/cares/config/darwin/ares_config.h b/deps/cares/config/darwin/ares_config.h
index 6187024a3a..19777617ca 100644
--- a/deps/cares/config/darwin/ares_config.h
+++ b/deps/cares/config/darwin/ares_config.h
@@ -329,7 +329,14 @@
 #define HAVE_SYS_PARAM_H 1
 
 /* Define to 1 if you have the <sys/random.h> header file. */
+/* nodejs-mobile: the iOS SDK has no <sys/random.h>; c-ares falls back to
+ * arc4random_buf on Apple, so guard this define to macOS only. */
+#if defined(__APPLE__)
+#  include <TargetConditionals.h>
+#endif
+#if !defined(TARGET_OS_IPHONE) || !TARGET_OS_IPHONE
 #define HAVE_SYS_RANDOM_H 1
+#endif
 
 /* Define to 1 if you have the <sys/select.h> header file. */
 #define HAVE_SYS_SELECT_H 1
